### DISCUSSION TITLE
Lesson 4: AttributeError: 'Eth' object has no attribute 'send_raw_transaction'*

### DISCUSSION THREAD

**Discussion by m090009** [smartcontractkit/full-blockchain-solidity-course-py#1388] [Q&A]
Platform: Windows
Web3 version: 5.12.2
Python: Python 3.7.1

Hi there, everything is working fine, however, when I go to sign my transaction a weird error appears in this line 

` tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)`

**  File "deploy.py", line 68, in <module>
    `tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)`
AttributeError: 'Eth' object has no attribute 'send_raw_transaction'**

which means that the web3.eth doesn't have the method `send_raw_transaction`

I reinstalled web3, installed an earlier version, but to no avail

**Comment by cromewar** [comment]
Hello @m090009 
According to the documentation: https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.sign_transaction
you are sending the transaction correctly, however is not working so it might be an indentation problem or something related around that line, so we'll need more info on this in order to help you.

Please give a look to #980 .
